data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/TemporalTypeHandler.java

```diff
@@ -117,8 +122,16 @@ private Long handleDateTime(final String dateTimeStr) {
     // Binlog reader converts temporal fields to epoch millis
     // The Binlog reader is set with EventDeserializer.CompatibilityMode.DATE_AND_TIME_AS_LONG
     private Long parseDateTimeStrAsEpochMillis(final String dateTimeStr) {
-        // Try parsing as Unix timestamp first
+        // Try parsing as Unix timestamp
         try {
+            // Date Time field in OpenSearch when represented as long value should be in milliseconds since the epoch
+            // If the value is already in microseconds (length > 13) convert to milliseconds
+            if (dateTimeStr.length() > 13) {
```

Comment:
So if a column of TIME type contains microseconds, then in the binlog event we get, it will show up as an epoch in microseconds?